code/model/modelcollide.cpp

```diff
@@ -488,6 +491,60 @@ void model_collide_tmappoly(ubyte * p)
 }
 
 
+// Textured Poly (2)
+// +0      int         id
+// +4      int         size
+// +8      vec3d      normal
+// +20     int         nverts
+// +24     int         tmap_num
+// +28     nverts*(model_tmap2_vert) vertlist (n,u,v)
+void model_collide_tmap2poly(ubyte* p) {
+	uint i;
+	uint nv;
+	uv_pair uvlist[TMAP_MAX_VERTS];
+	vec3d* points[TMAP_MAX_VERTS];
+	model_tmap_vert* verts;
+
+	Assert(Mc_pm->version >= 2300);
+
+	nv = uw(p + 20);
+
+	if (nv <= 0)
+		return;
+
+	if (nv > TMAP_MAX_VERTS) {
+		Error(LOCATION, "Model contains TMAP2 chunk with more than %d vertices!", TMAP_MAX_VERTS);
+		return;
+	}
+
+	int tmap_num = w(p + 24);
```

Comment:
Style nitpick. Please declare all variables on the first lines of the function (C style), or declare them as you need them, not a mixture of both.